arch/cortex-m/src/lib.rs

```diff
@@ -10,9 +10,6 @@
 
 use core::fmt::Write;
 
-#[cfg(all(target_arch = "arm", target_os = "none"))]
-use core::arch::global_asm;
```

Comment:
I removed these conditional imports while I was going over the code. Conditional imports are a great recipe for producing warnings or errors that can easily be avoided by just referring to items using their fully qualified name, or moving imports into the respective functions themselves.